index.bs

```diff
@@ -2893,8 +2950,255 @@ Note: Future variations of this specification will retain the invariant that
+execution context=].
+
+<div algorithm>
+
+The <dfn>stack trace for an exception</dfn> with an exception, or a [=Completion
+Record=] of type <code>throw</code>, |exception|, is given by:
+
+1. If |exception| is a value that has been thrown as an exception, let |record|
+   be the [=Completion Record=] created to throw |exception|. Otherwise let
+   |record| be |exception|.
+
+1. Let |stack| be the [=list of stack frames=] corresponding to execution at the
+   point |record| was created.
+
+1. Return [=construct a stack trace=] given |stack|.
+
+</div>
+
+#### The script.Target type #### {#type-script-Target}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl remote-cddl">
+RealmTarget = {realm: Realm};
+
+ContextTarget = {context: BrowsingContext}
+
+Target = (
+  RealmTarget //
+  ContextTarget
+);
+</pre>
+
+The <code>Target</code> type represents a value that is either a
+<code>script.Realm</code> or a <code>browsingContext.BrowsingContext</code>. This
+is useful in cases where a context identifier can stand in for the realm
+associated with the context's active document.
+
+<div algorithm>
+To <dfn>get a realm from a target</dfn> given |target|:
+
+1. If |target| matches the <code>ContextTarget</code> production:
+
+  1. Let |context| be the result of [=trying=] to [=get a browsing context=]
+     with |target|["<code>context</code>"].
+
+  1. Let |document| be |context|'s [=active document=].
+
+  1. Let |environment settings| be the [=environment settings object=] whose
+     [=responsible document=] is |document|.
+
+  1. Let |realm| be |environment settings|' [=realm execution context=]'s Realm component.
+
+1. Otherwise:
+
+  1. Assert: |target| matches the <code>RealmTarget</code> production.
+
+  1. Let |realm id| be the value of the <code>realm</code> field of |target|.
+
+  1. Let |realm| be [=get a realm=] given |realm id|.
+
+1. Return [=success=] with data |realm|
+
+Issue: This has the wrong error code
+</div>
+
+
 ### Commands ### {#module-script-commands}
 
+#### The script.evaluate Command ####  {#command-script-evaluate}
+
+The <dfn export for=commands>script.evaluate</dfn> command evaluates a provided
+script in a given realm. For convenience a browsing context can be provided in
+place of a realm, in which case the realm used is the realm of the browsing
+context's active document.
+
+The method returns the value of executing the provided script, unless it returns
+a promise and <code>awaitPromise</code> is true (which is the default), in which
+case the resolved value of the promise is returned.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      ScriptEvaluateCommand = {
+        method: "script.evaluate",
+        params: ScriptEvaluateParameters
+      }
+
+      ScriptEvaluateParameters = {
+        expression: text,
+        ?awaitPromise: bool,
+        target: Target,
+      }
```

Comment:
Does it matter in which order types are defined? In other places sub types are  defined first. Also sort alphabetically?